lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -34,25 +34,33 @@ def authorize(amount, payment_method, options={})
         params[:email] = options[:email] || 'unknown@email.com'
         params[:createAlias] = false
 
-        commit("authorize", params)
+        commit("authorize", params, options)
       end
 
       def capture(authorization, options={})
         params = {}
+        options[:id_unico] = parse_auth(authorization)[1]
         add_auth_order_id(params, authorization, options)
-        commit("deposit", params)
+        commit("deposit", params, options)
       end
 
       def void(authorization, options={})
         params = {}
         add_auth_order_id(params, authorization, options)
-        commit("void", params)
+        commit("void", params, options)
       end
 
       def refund(amount, authorization, options={})
+        # Attempt RefundSingleTransaction if unsettled
         params = {}
+        params[:amount] = amount(amount)
         add_auth_order_id(params, authorization, options)
-        commit("cancelDeposit", params)
+        response = commit("cancelDeposit", params, options)
+        return response if response.success? || parse_auth(authorization).length == 1
```

Comment:
I'm not sure I follow this conditional. So we'd return a failed `cancelDeposit` response if the authorization used for the refund didn't have one of the two ID numbers? Could you clarify why?

Reply:
We can't process a refund without the id that is the second part of the ID numbers. Trying to do so gives a new error. I felt it wasn't worth making the second call since we don't have the key piece of information we would need.

Reply:
Gotcha, that make sense.